src/FSharpLint.Console/Sarif.fs

```diff
@@ -0,0 +1,110 @@
+﻿module internal Sarif
+
+open System
+open System.IO
+open Microsoft.CodeAnalysis.Sarif
+open Microsoft.CodeAnalysis.Sarif.Writers
+open FSharpLint.Framework
+open FSharpLint.Console.Output
+
+let writeReport (results: Suggestion.LintWarning list) (codeRoot: string option) (report: string) (logger: IOutput) =
+    try
+        let codeRoot =
+            match codeRoot with
+            | None -> Directory.GetCurrentDirectory() |> Uri
+            | Some root -> Path.GetFullPath root |> Uri
+
+        // Construct full path to ensure path separators are normalized.
+        let report = Path.GetFullPath report
+        // Ensure the parent directory exists
+        let reportFile = FileInfo(report)
+        reportFile.Directory.Create()
+
+        let driver = 
+            ToolComponent(
+                Name = "FSharpLint.Console",
+                InformationUri = Uri("https://fsprojects.github.io/FSharpLint/"),
+                Version = string<Version> (System.Reflection.Assembly.GetExecutingAssembly().GetName().Version)
+            )
+
+        let tool = Tool(Driver = driver)
+        let run = Run(Tool = tool)
+
+        use sarifLogger =
+            new SarifLogger(
```

Comment:
The CI run against .NET 10 has failed here with a self-lint error about 'new' being spurious

<img width="794" height="104" alt="image" src="https://github.com/user-attachments/assets/118ad67d-2633-491d-a0ec-d0565ed66889" />

But i'm not sure why that's complaining as ```SarifLogger``` is a disposable type

Reply:
but the strange thing is that this is happening in the Net10 CI job, not in the main (Net9) one's SelfCheck step

Reply:
> but the strange thing is that this is happening in the Net10 CI job, not in the main (Net9) one's SelfCheck step

oh wait, you're right, I guess the Net10 CI job doesn't have a restore, so that's why; maybe we should make the rule RedundantNewKeyword never fire when analysing an untyped tree?

Reply:
10 I think.
However - looking at the CI output in more detail I'm not sure if the .NET 8 run is actually doing the lint - 'inline source' doesn't look right? 
<img width="878" height="190" alt="image" src="https://github.com/user-attachments/assets/e8e6ae31-8d1d-4a3b-8475-25ef9975cf41" />